### ISSUE TITLE
Tabular listings should have sort functionality on columns & default sorting

### ISSUE THREAD

**Issue by fpigeonjr** [HHS/OPRE-OPS#976]
### User Story
As a COR or team lead, when viewing tabular listings, I want to be able to see the information sorted in a way that's helpful to me in the moment to increase my efficiency in using OPS

### Acceptance Criteria
- [x] User is able to sort all tables of data (CANs, BLs, agreements, etc..) on any column and in either direction
- [x] The sorting should be mindful of and respect pagination and sort across the full set
- [x] The sorting should still respect any filters in place
- [x] Default sorting for the Agreement List should be the Agreement Name ascending
- [x] Default sorting for the CANs should be the CAN number ascending
- [x] Default sorting for the BL list should be based on its Agreement title and then by associated services component ascending

#### Dev
- [ ] add useSortableData hook to the `PreviewTable` component
- ref: Portfolio table

Definition of Done Checklist

- [ ] UI works as designed (UX team)
- [ ] Usability validated (UX team)
- [x] PR(s) have been merged to main
- [x] Design/tech debt eliminated
- [x] New design/tech debt documented (if applicable)
- [x] Build process updated
- [x] Documentation updated or added
- [x] Feature flags/toggles created

### Additional Context
* its intended to work the same as the add budget lines workflow where that already works

**Comment by kimschulke** [comment]
@fpigeonjr didnt we do this already?

**Comment by jonnalley** [comment]
@kimschulke no. it was de-prioritized over some of the other urgent work that is ongoing and will need some re-work given what's going on with 3648.

**Comment by fpigeonjr** [comment]
@jonnalley is correct. Once we get 3648 wrapped up, hope to return to this story hoping that the recent refactors don't affect the work already completed but that is a non-zero probability.

**Comment by kimschulke** [comment]
@rajohnson90 does this include sorting on the agreement details page - budget lines tables within each services component? We had a few people comment on the somewhat random sorting of budget lines and wanted to see if this work will improve that. Thank you!

**Comment by jonnalley** [comment]
@kimschulke It's awaiting review by the other developers before merging in and deployment out to dev/staging.

**Comment by fpigeonjr** [comment]
Great job with this Reed. I know a lot of work went into this. Sorting felt snappy and performant. Keyboard navigation worked well overall. Testing instructions were well documented. Here our some observations from testing in `localdev`:

- portfolio 1 w/ FY 2043 ✅
- agreements ❌
	- Type sort didn't seem to be working 
- agreement budget lines ✅
- budgetlines list ❌
	- Services Components seems off 
- CANs list ✅
	- keyboard nav only works on Funding Received and Available Budget
	- no hover indication on other rows // adding button element may solve this
	- arrow colors are off, in other tables they are black
- Review Table ✅
	- alignment is off on BLI ID column and other column headers
- Diff Table ✅

**Comment by fpigeonjr** [comment]
## Type sort on Agreements list

![Image](https://github.com/user-attachments/assets/b06c787f-dc54-4c66-8c27-4a316a5b7aa6)

## Services Component sort on Budget Lines list

<img width="321" alt="Image" src="https://github.com/user-attachments/assets/5e1128b4-e3a7-4412-8355-c5086ce8aa59" />

**Comment by fpigeonjr** [comment]
question for UX team: @kimschulke @andressacurvelo @npage123
should applying a filter bring you back to the first page of results?
- if I am page 9 of results and set a filter 
- should we go to page 1 or stay on page 9

**Comment by npage123** [comment]
Sort should encompass entire data set across all pages and not sort within each page respectively. Thinking the most common pattern would be user sorting on page 1 and then navigating to the desired page. But if the user navigates to page 9 and they sort, they would stay on page 9 but the page's results would update the total dataset and they can then navigate to whatever page.

**Comment by fpigeonjr** [comment]
thx @npage123, that is the current behavior in #3599.

**Comment by jonnalley** [comment]
@andressacurvelo @npage123 @kimschulke This is ready for review in dev & staging

**Comment by kimschulke** [comment]
This is awesome! But I received a "something went wrong error" when testing on the agreements list and trying to sort by agreement total column @rajohnson90

**Comment by fpigeonjr** [comment]
I am seeing this on Agreement list sort on **Agreement Totals** on staging

`https://stg.ops.opre.acf.gov/api/v1/agreements/?sort_conditions=AGREEMENT_TOTAL&sort_descending=true`
Request Method
GET
Status Code
500 Internal Server Error

**Comment by jonnalley** [comment]
@kimschulke that and other issues fixed, check it out again at your convenience and please do check off DoD items when ready please.  cc @andressacurvelo @npage123

**Comment by npage123** [comment]
In starting my review of the sorting functionality on the portfolio pages I've come across this with a couple columns not sorting correctly. 

https://github.com/user-attachments/assets/00df3b1b-0bfe-431a-b5b4-7ee8c990dc78

**Comment by npage123** [comment]
Here's another finding on the agreements page. It seems related to rows with TBD as a value within them, not necessarily it being the value in the column being sorted.

https://github.com/user-attachments/assets/c3ebc2cc-9490-40e0-b77d-7351325ebd72

**Comment by npage123** [comment]
The TBD strikes again!

https://github.com/user-attachments/assets/7e085493-3759-4108-a266-ebf0ff7c7873

**Comment by npage123** [comment]
Not sure this is related to sorting but an error received after filtering by a specific portfolio and then clicking on different columns to sort.

https://github.com/user-attachments/assets/c5852262-5494-4da2-8553-86ac42f0e2cd

**Comment by npage123** [comment]
Hmmm, I'm still getting TBDs as not getting sorted. And if it helps, when I filter by portfolio and then sort, it seems the error is always triggered by me sorting on the CAN column.

**Comment by npage123** [comment]
Upon Reed explaining the row with TBD is actually the large amount used to test a previous data issue and it just isn't loading due to it being such a large number and all the BL data, makes sense to me that that isn't a bug. 

A few other things I noticed:
1. Looking into the error occurring after filtering the BL list, it seems the trigger is always sorting by the CAN column.
2. Also, not seeing the default sorting applied in the AC if that is still a part of this issue.

**Comment by jonnalley** [comment]
closing for now based on Nate's comment above.
